badgecheck/tasks/graph.py

```diff
@@ -21,11 +23,22 @@ def fetch_http_node(state, task_meta):
             return task_result(message='Successfully fetched image from {}'.format(url))
         return task_result(success=False, message="Response could not be interpreted from url {}".format(url))
 
-    # TODO: Add JSON-LD COMPACTION
     actions = [add_task(JSONLD_COMPACT_DATA, data=data, node_id=url)]
     return task_result(message="Successfully fetched JSON data from {}".format(url), actions=actions)
 
 
 def jsonld_compact_data(state, task_meta):
     # TODO: Cache-friendly JSON-LD compaction into the Open Badges context
-    pass
+    input_data = json.loads(task_meta.get('data'))
+
+    node_id = task_meta.get('node_id')
+    options = {'documentLoader': CachableDocumentLoader(cachable=task_meta.get('use_cache', True))}
+
+    result = jsonld.compact(input_data, OPENBADGES_CONTEXT_URI_V2, options=options)
```

Comment:
It's actually fine to use a context in a badge object other than what we know as `OPENBADGES_CONTGEXT_URI_V2`. That's a feature, not a bug! (We'll get to a couple cases where it's going to be difficult in the future around Extensions, but cross that bridge when we come to it). For example, if the developer uses a French context, we want to compact it into the context our code is written against instead. If they have a complex context like `["https://w3id.org/openbadges/v2", {"myCustomTerm1": "http://a.com/1"}`, we want to see the full IRI for that term instead of its compact property name, so removing that variance by compaction is the right step.

We want to compact the input from whatever context it uses (might be ours, might be another) in order to:
1) normalize the JSON into what we can work with as just plain JSON and dicts without worrying about what property names are used.